frontend/src/app/App.js

```diff
@@ -12,40 +12,52 @@ import { CreateMatchPage } from './pages/CreateMatch'
 import { Profile } from './pages/Profile'
 import { AddPlayerPage } from './pages/AddPlayer'
 import { MatchListPage } from './pages/MatchList'
-import { Button } from '../styles/blocks'
+import { Button, Subtitle } from '../styles/blocks'
+import { isGameSelected } from './modules/games/games-selectors'
+import { connect } from 'react-redux'
 
-export class App extends Component {
+export class AppComponent extends Component {
   render() {
-    const { match: { url } } = this.props
+    const { match: { url, params: { gameName } }, isGameSelected, gameNotFound } = this.props
     const createMatch = () => {
       this.props.history.push(`${url}${ROUTES.CREATE_MATCH}`)
     }
     const constructUrl = relativePath => `${url}${relativePath}`
     return (
       <>
         <Header>
-          <Button onClick={createMatch}>Add Match</Button>
+          { isGameSelected? <Button onClick={createMatch}>Add Match</Button> : null }
         </Header>
         <Container>
-          <Switch>
-            <Route exact path={constructUrl(ROUTES.LEADERBOARD)}>
-              <LeaderboardPage constructUrl={constructUrl} />
-            </Route>
-            <Route exact path={constructUrl(ROUTES.CREATE_MATCH)}>
-              <CreateMatchPage constructUrl={constructUrl} />
-            </Route>
-            <Route exact path={constructUrl(ROUTES.ADD_PLAYER)} component={AddPlayerPage} />
-            <Route exact path={constructUrl(ROUTES.MATCH_LIST)}>
-              <MatchListPage constructUrl={constructUrl} />
-            </Route>
-            <Route exact path={constructUrl(ROUTES.PROFILE)} component={Profile} />
-            <Route>
-              <Dashboard constructUrl={constructUrl} />
-            </Route>
-          </Switch>
+          { isGameSelected
+            ? <Switch>
+              <Route exact path={constructUrl(ROUTES.LEADERBOARD)}>
+                <LeaderboardPage constructUrl={constructUrl} />
+              </Route>
+              <Route exact path={constructUrl(ROUTES.CREATE_MATCH)}>
+                <CreateMatchPage constructUrl={constructUrl} />
+              </Route>
+              <Route exact path={constructUrl(ROUTES.ADD_PLAYER)} component={AddPlayerPage} />
+              <Route exact path={constructUrl(ROUTES.MATCH_LIST)}>
+                <MatchListPage constructUrl={constructUrl} />
+              </Route>
+              <Route exact path={constructUrl(ROUTES.PROFILE)} component={Profile} />
+              <Route>
+                <Dashboard constructUrl={constructUrl} />
+              </Route>
+            </Switch>
+            : <Subtitle>{ gameNotFound? `Game '${gameName}' was not found!` : 'Loading...' } </Subtitle>
+          }
         </Container>
         <Footer />
       </>
     )
   }
 }
+
+const mapStateToProps = state => ({
+  isGameSelected: isGameSelected(state),
+  gameNotFound: state.gameNotFound,
```

Comment:
2 questions: 
1. isn't gameNotFound complementary to isGameSelected? Or can you select games that are not found?
2. usually we don't sue negative in variable names. it's easier to understamd variable isGameFound that gameNotFound

Reply:
1. the game may not be selected and may be still found though... gameNotFound as true means that the selection failed...

Reply:
2. because of 1 it would be more confusing because isGameFound true doesn't necessary mean that it actually is found but could be that it's was not yet selected at all... probably the name is wrong at all

Reply:
fixed with enum here https://github.com/salsita/foosball-rating/pull/141/commits/8c9de1db3591e4a5cb8bbdf72d80fe0052794850

Reply:
reverted and fixed with https://github.com/salsita/foosball-rating/pull/141/commits/04c18104046efea46d420642ac764d846b2c69a9